impl/src/main/java/com/okta/sdk/impl/client/DefaultClientBuilder.java

```diff
@@ -391,6 +403,70 @@ public ClientBuilder setPrivateKey(String privateKey) {
         return this;
     }
 
+    @Override
+    public ClientBuilder setPrivateKey(Path privateKeyPath) {
+        if (isOAuth2Flow()) {
+            Assert.notNull(privateKeyPath, "Missing privateKeyFile");
+            this.clientConfig.setPrivateKey(getFileContent(privateKeyPath));
+        }
+        return this;
+    }
+
+    @Override
+    public ClientBuilder setPrivateKey(InputStream privateKeyStream) {
+        if (isOAuth2Flow()) {
+            Assert.notNull(privateKeyStream, "Missing privateKeyFile");
+            this.clientConfig.setPrivateKey(getFileContent(privateKeyStream));
+        }
+        return this;
+    }
+
+    @Override
+    public ClientBuilder setPrivateKey(PrivateKey privateKey) {
+        if (isOAuth2Flow()) {
+            Assert.notNull(privateKey, "Missing privateKeyFile");
+            String encodedString = "-----BEGIN PRIVATE KEY-----\n";
```

Comment:
RSA private key usually has different headers `-----BEGIN RSA PRIVATE KEY-----` & `-----END RSA PRIVATE KEY` . Please check if this works with other key types we support (RSA, EC etc).

Reply:
Updated